### ISSUE TITLE
MySQL state=import does not fail when given wrong file type

### ISSUE THREAD

**Issue by makmanalp** [ansible/ansible#7195]
##### Issue Type:

“Bug Report”
##### Ansible Version:

ansible 1.5.4
##### Environment:

N/A
##### Summary:

Accidentally put in a gzip file with a .bz2 extension as the db dump. Would have expected an error. Instead it continued on as if the import worked, and went on to notify the next task.
##### Steps To Reproduce:

Import task like:

```
- name: import atlas db
  mysql_db: name=atlas state=import target=/vagrant_shared/atlas_dump.sql.bz2
```

where the .bz2 file is actually a gzip.
##### Expected Results:

Stop with error since import could not have worked.
##### Actual Results:

(Verbose output)

```
TASK: [db | import atlas db] **************************************************
<127.0.0.1> ESTABLISH CONNECTION FOR USER: vagrant
<127.0.0.1> REMOTE_MODULE mysql_db name=mysql state=import target=/vagrant_shared/atlas_dump.sql.bz2
<127.0.0.1> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/Users/makmana/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'Port=2222', '-o', 'IdentityFile=/Users/makmana/.vagrant.d/insecure_private_key', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=vagrant', '-o', 'ConnectTimeout=10', '127.0.0.1', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1398708205.3-195490876216141 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1398708205.3-195490876216141 && echo $HOME/.ansible/tmp/ansible-tmp-1398708205.3-195490876216141'"]
<127.0.0.1> PUT /var/folders/kx/pl6ftk6j01719wf22jx0v4whbptcsv/T/tmpIV7J3v TO /home/vagrant/.ansible/tmp/ansible-tmp-1398708205.3-195490876216141/mysql_db
<127.0.0.1> EXEC ['ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/Users/makmana/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'Port=2222', '-o', 'IdentityFile=/Users/makmana/.vagrant.d/insecure_private_key', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=vagrant', '-o', 'ConnectTimeout=10', '127.0.0.1', "/bin/sh -c '/usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1398708205.3-195490876216141/mysql_db; rm -rf /home/vagrant/.ansible/tmp/ansible-tmp-1398708205.3-195490876216141/ >/dev/null 2>&1'"]
changed: [default] => {"changed": true, "db": "mysql", "item": "", "msg": ""}

```

**Comment by Jmainguy** [comment]
This has been fixed by pull request #6766 . Suggest closing this out.

**Comment by makmanalp** [comment]
:+1: Thanks!
